packages/neon-api/src/notifications/class.ts

```diff
@@ -0,0 +1,49 @@
+import { logging, settings } from "@cityofzion/neon-core";
+import WebSocket from "isomorphic-ws";
+import { NotificationMessage } from "./responses";
+const log = logging.default("api");
+
+export class Notifications {
+  private url: string;
+
+  public get name(): string {
+    return `Notifications[${this.url}]`;
+  }
+
+  private websocketSubscriptions: Map<string | null, WebSocket[]>;
+
+  public constructor(url: string) {
+    this.url = settings.networks[url]?.extra?.notifications ?? url;
+    this.websocketSubscriptions = new Map<string | null, WebSocket[]>();
+    log.info(`Created Notifications Provider: ${this.url}`);
+  }
+
+  public subscribe(

```

Comment:
Regarding that, I'm not really sure when to write a description for a method and when not to. Looking at the current source the only pattern I can see is that functions with lots of lines are generally annotated, is there any heuristic in place for that?

Reply:
generally I would annotate the public api. So it would be the constructor, subscribe and unsubscribe here. Unsubscribe might be pretty straight forward so no need but the constructor and subscribe would benefit from some description to indicate how to use. For example, indicating that you can subscribe to all by not passing an argument.

Reply:
Makes sense, I thought that would be the case but when I looked at the code of the neoCli provider there was only a single function annotated, an internal one not directly exposed. But I can see that the functions there have more descriptive names so it would make sense not to annotate them.